enclave-runtime/src/tests.rs

```diff
@@ -275,8 +275,20 @@ fn test_create_block_and_confirmation_works() {
 	.unwrap();
 
 	// when
-	let execution_result = top_pool_executor
-		.execute_trusted_calls(&latest_parentchain_header(), &shard, MAX_TRUSTED_OPS_EXEC_DURATION)
+	let top_pool_calls = top_pool_executor.get_trusted_calls(&shard).unwrap();
+	let execution_result = stf_executor
+		.execute_timed_calls_batch_without_permanent_mutation::<Block, _>(
+			&top_pool_calls,
+			&latest_parentchain_header(),
+			&shard,
+			MAX_TRUSTED_OPS_EXEC_DURATION,
+			|s| {
+				let mut sidechain_db = SidechainDB::<SignedBlock, SgxExternalities>::new(s);
+				sidechain_db
+					.set_block_number(&sidechain_db.get_block_number().map_or(1, |n| n + 1));
+				sidechain_db.ext
+			},
+		)
```

Comment:
So now we have a top pool executor that doesn't really execute anything anymore and in addition we need the `stf_executor` to do the execution?